src/servers/src/http/event.rs

```diff
@@ -147,6 +147,51 @@ where
     }
 }
 
+#[axum_macros::debug_handler]
+pub async fn query_pipeline(
+    State(state): State<LogState>,
+    Extension(mut query_ctx): Extension<QueryContext>,
+    Query(query_params): Query<LogIngesterQueryParams>,
+    Path(pipeline_name): Path<String>,
+) -> Result<GreptimedbManageResponse> {
+    let start = Instant::now();
+    let handler = state.log_handler;
+    ensure!(
+        !pipeline_name.is_empty(),
+        InvalidParameterSnafu {
+            reason: "pipeline_name is required in path",
+        }
+    );
+
+    let version_str = query_params.version.context(InvalidParameterSnafu {
+        reason: "version is required",
+    })?;
+
+    let version = to_pipeline_version(Some(&version_str)).context(PipelineSnafu)?;
+
+    query_ctx.set_channel(Channel::Http);
+    let query_ctx = Arc::new(query_ctx);
+
+    let result = handler
+        .get_pipeline(&pipeline_name, version, query_ctx)
+        .await;
+
+    result
+        .map(|pipeline: Arc<pipeline::Pipeline>| {
+            let p = serde_json::to_value(&pipeline).unwrap();
```

Comment:
Can we output the pipeline in pretty yaml string? It will be easy to read.

Reply:
> Can we output the pipeline in pretty yaml string?

Do you have any example？ If I found the pretty method then I will update the commit